src/file_pipe_log/pipe.rs

```diff
@@ -18,13 +18,76 @@ use crate::metrics::*;
 use crate::pipe_log::{FileBlockHandle, FileId, FileSeq, LogQueue, PipeLog};
 use crate::{Error, Result};
 
-use super::format::{FileNameExt, Version};
+use super::format::{FileNameExt, LogFileContext, Version};
 use super::log_file::{build_file_reader, build_file_writer, FileHandler, LogFileWriter};
 
 struct FileCollection<F: FileSystem> {
+    /// first seq of active files
     first_seq: FileSeq,
+    /// final seq of active files
     active_seq: FileSeq,
+    /// first seq of stale files
+    first_stale_seq: FileSeq,
```

Comment:
`active_seq` can be removed because it can be deduced from `fds.len()`.
```suggestion
    /// Sequence number of the first file.
    first_seq: FileSeq,
    /// Sequence number of the first file that is in use.
    first_seq_in_use: FileSeq,
```